modulus/distributed/manager.py

```diff
@@ -533,6 +542,7 @@ def create_groups_from_config(
     def cleanup():
         """Clean up distributed group and singleton"""
         # Destroying group.WORLD is enough for all process groups to get destroyed
-        dist.barrier()  # just make sure that no process hangs
-        dist.destroy_process_group()
+        if DistributedManager().distributed:
+            dist.barrier()  # just make sure that no process hangs
```

Comment:
it is good practice to specify a device id here. something like: `dist.barrier(device_ids=[<local_device_index>])`. Otherwise barriers could cause issues if more than one device is present and if there are no cuda contexts for these devices.

Reply:
Agreed, added that and guarded with `if torch.cuda.is_available()`